Python/pylifecycle.c

```diff
@@ -762,18 +762,32 @@ pycore_init_builtins(PyThreadState *tstate)
     }
     interp->builtins = Py_NewRef(builtins_dict);
 
-    PyObject *isinstance = PyDict_GetItem(builtins_dict, &_Py_ID(isinstance));
-    assert(isinstance);
+    PyObject *isinstance;
+    if (PyDict_GetItemRef(builtins_dict, &_Py_ID(isinstance), &isinstance) != 1) {
+        goto error;
+    }
     interp->callable_cache.isinstance = isinstance;
-    PyObject *len = PyDict_GetItem(builtins_dict, &_Py_ID(len));
-    assert(len);
+    Py_DECREF(isinstance);
+
+    PyObject *len;
+    if (PyDict_GetItemRef(builtins_dict, &_Py_ID(len), &len) != 1) {
```

Comment:
There is no actually advantage over using `PyDict_GetItemWithError()`, because you decref the reference immediately after getting it.

And why not use `_PyEval_GetBuiltin()`?

Reply:
Right, I'm trying to see where PyDict_GetItemRef() fits better: that's why I propose to add PyDict_ContainsString() in issue #108314 :-) I updated my PR to use PyDict_GetItemWithError().

> And why not use _PyEval_GetBuiltin()?

Hum, I prefer to stick to the current code, since we are initializing Python which is a half-broken state.